tests/HTML5ValueTest.php

```diff
@@ -73,4 +75,22 @@ public function testMixedNewlines()
             'Newlines get converted'
         );
     }
+
+    public function testShortcodeValue()
+    {
+        ShortcodeParser::get('default')->register(
+            'test_shortcode',
+            function () {
+                return 'bit of test shortcode output';
+            }
+        );
+        $content = DBHTMLText::create('Test', ['shortcodes'=>true])
+            ->setValue('<p>Some content with a [test_shortcode] and a <br /> followed by an <hr> in it.</p>')
```

Comment:
I think this test is testing 2 things (1. that the HTML parser moves `<hr>` out of its `<p>` and 2. that the shortcode is correctly parsed.

I don't think there's a need for the first test (the lib itself should be testing for this) but if you want to test it, it should be a separate test.

Reply:
No, you're right. It should be tested in the lib. And it shouldn't be an issue.
But neither should it feel the need to close void elements (nor do they in the lib tests).

The opportunity was there, I took it. It is valid in this test, and it is only testing that the `<hr>` doesn't render as `<hr></hr>`, I've simply left the comment to make clear why the tag order has rearranged in the expected output (for those who perhaps may not be aware).